src/server/worker/driver/driver.go

```diff
@@ -76,16 +249,128 @@ func lookupGroup(group string) (_ *user.Group, retErr error) {
 	return nil, fmt.Errorf("group %s not found", group)
 }
 
-func (a *APIServer) chunks(jobID string) col.Collection {
-	return col.NewCollection(a.etcdClient, path.Join(a.etcdPrefix, chunkPrefix, jobID), nil, &ChunkState{}, nil, nil)
+func (d *driver) WithCtx(ctx context.Context) Driver {
+	result := &driver{}
+	*result = *d
+	result.pachClient = result.pachClient.WithCtx(ctx)
+	return result
+}
+
+func (d *driver) Jobs() col.Collection {
+	return d.jobs
+}
+
+func (d *driver) Pipelines() col.Collection {
+	return d.pipelines
 }
 
-func (a *APIServer) merges(jobID string) col.Collection {
-	return col.NewCollection(a.etcdClient, path.Join(a.etcdPrefix, mergePrefix, jobID), nil, &MergeState{}, nil, nil)
+func (d *driver) Plans() col.Collection {
+	return d.plans
 }
 
-func (a *APIServer) downloadData(pachClient *client.APIClient, logger logs.TaggedLogger, inputs []*common.Input, puller *filesync.Puller, stats *pps.ProcessStats, statsTree *hashtree.Ordered) (_ string, retErr error) {
-	defer a.reportDownloadTimeStats(time.Now(), stats, logger)
+func (d *driver) Shards() col.Collection {
+	return d.shards
+}
+
+func (d *driver) Chunks(jobID string) col.Collection {
+	return col.NewCollection(d.etcdClient, path.Join(d.etcdPrefix, chunkPrefix, jobID), nil, &common.ChunkState{}, nil, nil)
+}
+
+func (d *driver) Merges(jobID string) col.Collection {
+	return col.NewCollection(d.etcdClient, path.Join(d.etcdPrefix, mergePrefix, jobID), nil, &common.MergeState{}, nil, nil)
+}
+
+func (d *driver) GetExpectedNumWorkers() (int, error) {
+	return ppsutil.GetExpectedNumWorkers(d.kubeClient, d.pipelineInfo.ParallelismSpec)
+}
+
+func (d *driver) NewSTM(ctx context.Context, cb func(col.STM) error) (*etcd.TxnResponse, error) {
+	return col.NewSTM(ctx, d.etcdClient, cb)
+}
+
+func (d *driver) WithProvisionedNode(
```

Comment:
Picking up the naming conversation, since the code moved around. I'd just call this `WithData`, since it basically just takes the `data` argument and checks it out into the filesystem, then calls `cb` and then cleans up the data.